src/codegen/generators/variable_generator.rs

```diff
@@ -78,13 +78,15 @@ impl<'ctx, 'b> VariableGenerator<'ctx, 'b> {
             let linkage =
                 if !variable.is_in_unit(location) { LinkageType::External } else { variable.get_linkage() };
             let global_variable =
-                self.generate_global_variable(variable, linkage).map_err(|err| match err.get_type() {
-                    "E072" | "E048" => {
-                        Diagnostic::error(format!("Cannot generate literal initializer for `{name}`."))
-                            .with_error_code("E041")
-                            .with_sub_diagnostic(err)
+                self.generate_global_variable(variable, linkage).map_err(|err| {
+                    match err.get_error_code() {
+                        "E072" | "E048" => {
+                            Diagnostic::new(format!("Cannot generate literal initializer for `{name}`."))
+                                .with_error_code("E041")
+                                .with_sub_diagnostic(err)
+                        }
```

Comment:
this looks unique :-)
can you add a comment that explains what were doing here, if you remember

Reply:
Yes, it is :) I'll try to document it better/ see if we can avoid it.